gnomad/utils/vep.py

```diff
@@ -741,61 +883,147 @@ def get_most_severe_consequence_for_summary(
+    protein-coding transcripts and determining the most severe consequence. If no
+    protein-coding transcripts are present, determine the most severe consequence for
+    all transcripts. If additional VEP consequences lists are requested, process those
+    lists in the order they appear in `csq_list_order`.
+
+    If `add_order_by_csq_list` is provided, additional ordering is applied to the
+    consequences in the list. The key is the name of the consequences list and the value
+    is the order of consequences, sorted from high to low impact. An example use of this
+    parameter is to prioritize PolyPhen consequences for protein-coding transcripts.
+
+    If `prioritize_loftee` is True, prioritize consequences with LOFTEE annotations, in
+    the order of `loftee_labels`, over those without LOFTEE annotations. If
+    `prioritize_loftee_no_flags` is True, prioritize LOFTEE consequences with no flags
+    over those with flags.
+
+    :param vep_expr: StructExpression of VEP consequences to get the most severe
+        consequence from.
+    :param csq_order: Order of VEP consequences, sorted from high to low impact. Default
+        is None, which uses the value of the `CSQ_ORDER` global.
+    :param loftee_labels: Annotations added by LOFTEE, sorted from high to low impact.
+        Default is None, which uses the value of the `LOFTEE_LABELS` global.
+    :param include_csqs: Whether to include all consequences for the most severe
+        consequence. Default is False.
+    :param prioritize_protein_coding: Whether to prioritize protein-coding transcripts
+        when determining the worst consequence. Default is True.
+    :param prioritize_loftee: Whether to prioritize consequences with LOFTEE annotations
+        over those without. Default is False.
+    :param prioritize_loftee_no_flags: Whether to prioritize LOFTEE annotated
+        consequences with no flags over those with flags. Default is False.
+    :param csq_list_order: Order of VEP consequences lists to be processed. Default is
+        ('transcript_consequences', 'regulatory_feature_consequences',
+        'motif_feature_consequences', 'intergenic_consequences').
+    :param add_order_by_csq_list: Dictionary of additional ordering for VEP consequences
+        lists. The key is the name of the consequences list and the value is the order
+        of consequences, sorted from high to low impact. Default is None.
+    :return: Table annotated with VEP summary annotations.
+    """
+    add_order_by_csq_list = add_order_by_csq_list or {}
+    loftee_labels = loftee_labels or LOFTEE_LABELS
+
+    # Create a struct with missing values for each VEP consequences list.
+    ms_csq_list_expr = hl.struct(
+        **{c: hl.missing(vep_expr[c].dtype) for c in csq_list_order if c in vep_expr}
     )
 
+    # Build the case expression to determine the most severe consequence.
+    ms_csq_expr = hl.case(missing_false=True)
```

Comment:
why use case expressions?

Reply:
Is there a reason not to? If you have a clearer solution, I'm happy to make modifications, just let me know what your thinking